Userland/Libraries/LibGUI/FileSystemModel.cpp

```diff
@@ -671,6 +671,9 @@ bool FileSystemModel::is_editable(const ModelIndex& index) const
 void FileSystemModel::set_data(const ModelIndex& index, const Variant& data)
 {
     VERIFY(is_editable(index));
+    if(data.to_string().is_whitespace() || data.to_string() == "." || data.to_string() == "..")
+        return; // Filename is illegal
```

Comment:
Seems that if just calling rename with "." or ".." crashes the Kernel, we should fix this in sys$rename or in VFS::rename on the kernel side.

Reply:
Yes, this should absolutely be fixed in the kernel and not here.
cc @MaxWipfli who touched path resolution recently.